pkg/apis/core/validation/validation.go

```diff
@@ -2909,6 +2935,9 @@ func ValidateVolumeMounts(mounts []core.VolumeMount, voldevices map[string]strin
 		if mnt.MountPropagation != nil {
 			allErrs = append(allErrs, validateMountPropagation(mnt.MountPropagation, container, fldPath.Child("mountPropagation"))...)
 		}
+		if !opts.AllowInvalidRecursiveReadOnlyMounts && mnt.RecursiveReadOnly != nil {
```

Comment:
Don't need the conditional `AllowInvalidRecursiveReadOnlyMounts` - if the field has a value at this point, we validate it.

Reply:
fixed